common/Resources/ti.internal/extensions/node/fs.js

```diff
@@ -0,0 +1,1320 @@
+ * @param {string} [options.encoding=null] encoding to use
+ * @param {string} [options.flag='r'] file system flag
+ * @returns {string|Buffer} string if encoding is specified, otherwise Buffer
+ */
+fs.readFileSync = (path, options) => {
+	options = mergeDefaultOptions(options, { encoding: null, flag: 'r' });
+
+	// TODO: What if user passed in a file stream themselves, what do we do then?!
+	const wasFileDescriptor = (typeof path === 'number');
+	const fileDescriptor = wasFileDescriptor ? path : fs.openSync(path, options.flag); // use default mode
+
+	const tiFileStream = streamForDescriptor(fileDescriptor);
+	// Just use our own API that reads full stream in
+	const buffer = Ti.Stream.readAll(tiFileStream);
+
+	// fs.closeSync if it was not originally a file descriptor
+	if (!wasFileDescriptor) {
+		fs.closeSync(fileDescriptor);
+	}
+
+	// TODO: trim buffer if we didn't read full size?
+
+	return encodeBuffer(options.encoding, buffer);
+};
+
+// TODO: fs.readlink(path[, options], callback)
+// TODO: fs.readlinkSync(path[, options])
+// TODO: fs.realpath(path[, options], callback)
+// TODO: fs.realpath.native(path[, options], callback)
+// TODO: fs.realpathSync(path[, options])
+// TODO: fs.realpathSync.native(path[, options])
+
+/**
+ * @param {string|Buffer|URL} oldPath source filepath
+ * @param {string|Buffer|URL} newPath destination filepath
+ * @param {errorCallback} callback async callback
+ */
+fs.rename = (oldPath, newPath, callback) => {
+	setTimeout(() => {
+		try {
+			fs.renameSync(oldPath, newPath);
+		} catch (e) {
+			callback(e);
+			return;
+		}
+		callback();
+	}, 1);
+};
+
+/**
+ * @param {string|Buffer|URL} oldPath source filepath
+ * @param {string|Buffer|URL} newPath destination filepath
+ */
+fs.renameSync = (oldPath, newPath) => {
+	const tiFile = getTiFileFromPathLikeValue(oldPath);
+	// src doesn't actually exist?
+	if (!tiFile.exists()) {
+		const err = noSuchFile('rename', oldPath);
+		err.message = `${err.message} -> '${newPath}'`;
+		err.dest = newPath;
+		throw err;
+	}
+
+	const destFile = getTiFileFromPathLikeValue(newPath);
+	if (destFile.isDirectory()) {
+		// dest is a directory that already exists
+		const err = illegalOperationOnADirectory('rename', oldPath);
+		err.message = `${err.message} -> '${newPath}'`;
+		err.dest = newPath;
+		throw err;
+	}
+
+	let tempPath;
+	if (destFile.isFile()) {
+		// destination file exists, we should overwrite
+		// Our APIs will fail if we try, so first let's make a backup copy and delete the the original
+		tempPath = path.join(fs.mkdtempSync(path.join(Ti.Filesystem.tempDirectory, 'rename-')), path.basename(newPath));
+		destFile.move(tempPath);
+	}
+
+	let success = false;
+	try {
+		success = tiFile.move(newPath);
+	} finally {
+		if (tempPath) {
+			// we temporarily copied the existing destination to back it up...
+			if (success) {
+				// move worked, so we can wipe it away whenever...
+				fs.unlink(tempPath, err => {});
```

Comment:
<!--
  0 failure: 
  1 warning:  common/Resources/...
  
  
  DangerID: danger-id-Danger;
  File: common/Resources/ti.internal/extensions/node/fs.js;
  Line: 892;
-->

- :warning: common/Resources/ti.internal/extensions/node/fs.js line 892 – 'err' is defined but never used. Allowed unused args must match /^_.+/u. (no-unused-vars)